src/frontends/onnx/frontend/src/editor.hpp

```diff
@@ -31,9 +31,15 @@ class ONNX_IMPORTER_API ONNXModelEditor final {
     ///        is parsed and loaded into the m_model_proto member variable.
     ///
     /// \param model_path Path to the file containing the model.
-    ONNXModelEditor(const std::string& model_path, frontend::ExtensionHolder extensions = {});
+    /// \param extensions Holder of ONNX front end extensinos. Default {}.
+    /// \param shared_obj Pointer to frontend library. Default nullptr.
+    ONNXModelEditor(const std::string& model_path,
+                    frontend::ExtensionHolder extensions = {},
+                    std::shared_ptr<void> shared_obj = nullptr);
```

Comment:
Make it optional as last one to be more compliant with current tests. If this parameter is not optional than a lot of changes is required in files:
- src/frontends/tests/onnx/onnx_editor.cpp
- src/frontends/tests/onnx/onnx_import_with_editor.in.cpp
- src/frontends/tests/onnx/onnx_import_with_editor.in.cpp
- src/frontends/tests/onnx/onnx_transformations.cpp

It looks like for these tests are not required to set `shared_object`.

Reply:
Significant amount of changes would/should be addressed in further PR to avoid overwhelming this PR diff view. 
If these tests don't need shared object handle then an empty one can be provided. Having this parameter obligatory enforces reflection in further development. Aim of this PR is to keep objects closed in controlled manner.
@tomdol, would you advise?